mkt/site/middleware.py

```diff
@@ -111,7 +111,7 @@ def process_request(self, request):
             if stored_ov_lang.lower() in settings.LANGUAGE_URL_MAP:
                 ov_lang = stored_ov_lang
 
-        if 'lang' in request.REQUEST:
+        if 'lang' in request.GET:
```

Comment:
Maybe we need `POST` here too. I'm thinking if you are doing a `POST` with a `lang` value it should set it like it does with a `GET`. Not sure if we have code depending on this behaviour though